### DISCUSSION TITLE
Welcome to muxp Discussions!

### DISCUSSION THREAD

**Discussion by nofaceinbook** [nofaceinbook/muxp#51] [General]
## 👋 Welcome!
You might start here discussions on muxp like new ideas for muxp features.
Issues with existing shall however still be an issue.

**Comment by Colins2** [comment]
Hi,
I built Gorgiok from your muxp file and it looks pretty good.
I do most of my testing and (X-Plane) flying in PNG and I am very curious why you picked Gorgiok.
I have 5 databases of PNG airports, the largest of which contains just under 600 airstrips and none of then have Gorgiok listed anywhere.
It does show on Google Earth and Bing maps but without any details of course. I think it maybe closed, looking at the surface.

Can you tell me where you got the info for this airport as there are several others that Missionary Bush Pilot flies to that don't appear in my lists either, so if there is a more comprehensive list somewhere, I'd like to get a copy.

Regards
Colin

**Reply by nofaceinbook** [reply]
Hi Colin,
the airport was created by RandomDude93 in X-Plane forum:
https://forums.x-plane.org/index.php?/files/file/67463-png-airfields-series-1/
I just supported him with the mesh.

**Comment by Colins2** [comment]
Excellent, thanks.
I've just downloaded his package and library - nice to see Goriok populated
:-)
The library will really come in useful as there isn't much in the other
available libraries that looks right.
Colin

**Comment by pinecone7** [comment]
Hey, in MUXP 0.4.0 is there a way to directly convert DSF to MUXP? I'm working on KCMH which has a DSF boundary right through the area that I'm working on.

**Comment by nofaceinbook** [comment]
I'm not sure if I got your question @pinecone7 . MUXP files just include information how to change a DSF file. So you can't convert DSF to MUXP. But I guess your question is about how to change the mesh if it affects multiple DSF files like for KCMH. For this you have to adapt all dsf files involved with multiple muxp files. In your case on muxp file to adpat 40/-82 and one for 39/-82. However both files might include the same commands like creating a profiled runway. But in your case it seems even more easy like one runway is completley north and the outher completeley in the south part. 
You need also to take care that the area definition in the muxp file must always stay in one dsf grid.  This does not work with automatic muxp file creation. But you could adapt the files within the editor. Hope that helps.

**Reply by pinecone7** [reply]
Thank you for your response. I managed to get it to work, but it gets stuck on "Processing cut_spline.segment.runway_0." Is there a specific thing that could cause this?
EDIT: It says "Failed to execute script muxp"

[muxp.log](https://github.com/nofaceinbook/muxp/files/7403968/muxp.log)

**Comment by nofaceinbook** [comment]
Looks like an error in muxp. Could you also send me the muxp file. Then I might search for the error....

**Reply by pinecone7** [reply]
[automuxed_airport_KCMH.txt](https://github.com/nofaceinbook/muxp/files/7413528/automuxed_airport_KCMH.txt)

Can't attach a MUXP so here's an equivalent .TXT. I edited the coordinates to avoid the DSF boundary. One thing I noticed is that the file says version 0.32 whereas the actual application is 0.4.0.

**Comment by nofaceinbook** [comment]
Hi @pinecone7 , 
well, using strip for this airport might not a good solution. The strip function is usually for a small landing strip (single runway) and not a big airport. If you use this for multiple runways you will probably end up with unwanted results as updates for the second runway will change the whole airport boundary area again. 
Also I did not test this function to work in case of a dsf grid border crossing the airport. Did you try TIN? Or what exactly do you want to achieve with the mesh update.
By the way you only should update the coordinates in the area definition. The other coordinates that cross the grid border could still stay in the muxp file.

**Comment by nofaceinbook** [comment]
I also just see that MUXP creates for strip only the profile for the first runway. As this runway is completely in the other grid as you have defined within the area, this can not work. So I really suggest to use the TIN option.

**Reply by pinecone7** [reply]
I'm trying TIN now, I'll update you in a minute.

**Comment by nofaceinbook** [comment]
And just releaized, that the version 0.4 of muxp does not include the newest functions I have implemented to handly tile borders :-( 
Need to do some checks here and will let you know.

**Reply by pinecone7** [reply]
With TIN and the corrected coordinates, it still gets stuck on "Processing cut_spline.segment.runway_0"

**Comment by nofaceinbook** [comment]
The new version is working with the muxp files. If you can run python source code from this github you can directly try.  As I'm currently quite busy it might take some days until I will create the new release including .exe file. The attached file is creating a runway profile for the runway south of the airport
[automuxed_airport_KCMH.muxp.txt](https://github.com/nofaceinbook/muxp/files/7421562/automuxed_airport_KCMH.muxp.txt)
.

**Comment by nofaceinbook** [comment]
Created a quick fix for the .exe which is not public visible. Can you access it: https://github.com/nofaceinbook/muxp/releases/tag/untagged-a1692193988fbb7273b6

**Reply by pinecone7** [reply]
When I open the objects in blender, all I see is this?

![image](https://user-images.githubusercontent.com/79820366/139146091-cbd2d929-2136-4364-b59d-d648f05efa80.png)

**Reply by nofaceinbook** [reply]
Mesh Editing is not easy in blender and many issues can happen. Especially I did not test yet re-import of adapted mesh directly at dsf tile boundary. 
However,  if you would like to continue you should read this: 

https://forums.x-plane.org/index.php?/files/file/69631-updating-x-plane-mesh-with-muxp-and-blender/

What do you like to change? Perhaps there are easier methods than with blender...

**Reply by pinecone7** [reply]
![A4FCEFF9-C77E-4D67-99BE-F949AF95BD66](https://user-images.githubusercontent.com/79820366/139254839-e1079f46-9c4e-4779-997a-7729cdc37d12.jpeg)

I’m trying to slightly depress the area of KCMH highlighted in red, to make that section of taxiway H look like the bridge it is in real life.

**Comment by nofaceinbook** [comment]
You could use the cut_path command with double_cut and smooth option, as defined in the muxp manual. The path follows the road and you can assign elevation for each coordinate of the road. I also used it for that purpose. If you do this in Google earth or WED you can directly click the coordinates for that.

**Reply by pinecone7** [reply]
![image](https://user-images.githubusercontent.com/79820366/139366215-00a17a41-342c-45a6-be84-b8e9ccb8ee2c.png)
![image](https://user-images.githubusercontent.com/79820366/139366310-834f2c7e-4131-4073-b209-55c06aaca96a.png)
[scenery_packs.txt](https://github.com/nofaceinbook/muxp/files/7438555/scenery_packs.txt)
[+39-083 airport_KCMH 1.0.txt](https://github.com/nofaceinbook/muxp/files/7438562/%2B39-083.airport_KCMH.1.0.txt)
[+40-083 airport_KCMH 1.0.txt](https://github.com/nofaceinbook/muxp/files/7438563/%2B40-083.airport_KCMH.1.0.txt)

Is something wrong? I put the two DSFs in "KCMH BRIDGE MESH PATCH."

**Comment by nofaceinbook** [comment]
Cut_path works different. You assign some 3d_coordinates on the road with elevation.  The width is in meters for your road (so more like 60 than 1) and then you use double_cut (also cut the mesh where elevation starts to drop down to the road) and distance where  this drop starts (like 30). But you don't cut in that case the area you marked but really a path with width.

If you want to do it with the area you marked it is also possible. Then you use two normal cut commands. One as you did where the drop starts with upper elevation and one cut for the road with lower elevation.

**Reply by pinecone7** [reply]
I'm making strides, but the boundary is very messy
![image](https://user-images.githubusercontent.com/79820366/139514181-71df0ea1-95e7-41aa-921d-803b2ab56a43.png)
![image](https://user-images.githubusercontent.com/79820366/139514145-73632f62-433a-438f-b227-40edf88f854f.png
[+39-083 airport_KCMH 1.0.txt](https://github.com/nofaceinbook/muxp/files/7445489/%2B39-083.airport_KCMH.1.0.txt)
[+40-083 airport_KCMH 1.0.txt](https://github.com/nofaceinbook/muxp/files/7445490/%2B40-083.airport_KCMH.1.0.txt)
)
If I place the cuts any closer to the boundary, I get an "index out of range" error.

**Comment by nofaceinbook** [comment]
I was not aware that this bridge is directly at the tile boundary. In that case cut_path is not the right command as it seems that it can not cross the boundary when the width is creating coordinates on the other side.
So I would suggest to do it the other way. Create normal cuts for upper and lower elevation. They shuld be able to cross the tile boundary. When I find some time I could las try....

**Comment by nofaceinbook** [comment]
This is what I meant with inner and outer cut. The muxp filles attached are running. However, the results look not  really convincing yet. Especially, if you don't create your own road elements or even custom scenery bridges you will not really get nice results. Such mesh/road updates are really complex.

[+39-083.airport_KCMH_cutpoly.1.0.muxp.txt](https://github.com/nofaceinbook/muxp/files/7449721/%2B39-083.airport_KCMH_cutpoly.1.0.muxp.txt)

[+40-083.airport_KCMH_cutpoly.1.0.muxp.txt](https://github.com/nofaceinbook/muxp/files/7449722/%2B40-083.airport_KCMH_cutpoly.1.0.muxp.txt)
![Inner OuterCut](https://user-images.githubusercontent.com/5694651/139600052-1d744d24-654e-4b1b-a1e2-5b88fd62e49b.JPG)

**Comment by nofaceinbook** [comment]
But looking to your screenshots it seems you exclude default roads and put ortho image over it, this could work. And if you cut also the area under the bridges and place your own object-bridges over the cut you could get really good results.

**Reply by pinecone7** [reply]
Added more points but the DSF boundary is still bad and one side isn't showing at all.

![image](https://user-images.githubusercontent.com/79820366/139770549-08f989e1-f7a0-4486-9e17-8ba43ee61a7f.png)
![image](https://user-images.githubusercontent.com/79820366/139770565-fdd8a1bb-0b6b-414b-aa0d-cbd1cdaff4c3.png)
![image](https://user-images.githubusercontent.com/79820366/139770684-20aac63b-cb37-4985-a95f-a81f4e7923d8.png
[+40-083.airport_KCMH_cutpoly.1.0.muxp.txt](https://github.com/nofaceinbook/muxp/files/7457052/%2B40-083.airport_KCMH_cutpoly.1.0.muxp.txt)
[+39-083.airport_KCMH_cutpoly.1.0.muxp.txt](https://github.com/nofaceinbook/muxp/files/7457053/%2B39-083.airport_KCMH_cutpoly.1.0.muxp.txt)
)

**Reply by pinecone7** [reply]
Got the other side to work, but not the boundary. Also, my Forkboy orthophotos aren't showing for the tiles that are edited. Is there a way to integrate these?

**Comment by nofaceinbook** [comment]
Hi @pinecone7 , sorry for late answer. I have currently many real problems to solve.
Good to hear that you get the other side to work. What was the issue? I did not see an issue with the muxp file but did not have time to test. And what do mean that the boundary is not working? 
One issue could be that you created too many points for the boundary. These many mesh triangles could eaily cause problems and they have also bad impact of the performance. You also will not recognize these fine grained changes from the plane. Would really just use 5 to 15 points for each polygon. 

Forkboy is not only creating orthophotos. This is a full mesh. So if you want to have your changes visible for Forkboys ortho-mesh you need to run muxp on these dsf files and not on X-Plane default mesh. One way to do this is by removing the following line from the muxp file. Then you will see a new menu where you can choose the mesh to adapt.

`source_dsf: DEFAULT`

Hope that helps.

**Reply by pinecone7** [reply]
The issue with the other side was that it was cutting the lower elevation before the higher elevation, so the higher elevation showed for the whole thing. The DSF boundary problem is shown below:

![image](https://user-images.githubusercontent.com/79820366/140445702-03b628c8-ac25-49ad-bd89-a74222b83386.png)
